### PR TITLE
Amqp interop based module.

### PR DESCRIPTION
**Pull request by makasim**
The PR add another AMQP module but based on [AMQP interop](https://github.com/queue-interop/queue-interop#amqp-interop). I tried to keep the module as close as possible to currently existing so It might be a replacement for it in future. 

There are several reasons for that:

* The code does not rely on an implementation but interoperable interfaces.
* The biggest advantage among others is that it would work with not only php-amqplib but amqp-ext, and bunny, or any amqp interop compatible libraries.
cleaner OO API.
* less code, fewer bugs.

### CODE REVIEW

File: src/Codeception/Module/AMQPInterop.php [hunk 1]
```diff
@@ -0,0 +1,416 @@
...
+            /** @var AmqpConnectionFactory $factory */
+            $factory = new $factoryClass([
+                'host' => $this->config['host'],
+                'port' => $this->config['port'],
+                'user' => $this->config['username'],
+                'pass' => $this->config['password'],
+                'vhost' => $this->config['vhost'],
+            ]);
+
+            $this->context = $factory->createContext();;
```
[the remark this answers is not in the document]
Comment (bscheshirwork) [reply src/Codeception/Module/AMQPInterop.php]: double semicolon

**Comment by DavertMik** [comment]
>  another AMQP module

Do we really need *another*? Yes, this one can be more stable and have fewer bugs but who knows?
Yeah, It's really cool that you have multiple backends supported. But currently, I'm not so happy to keep AMQP in core (but it can't be moved out before v3) and it has no meaning to keep two of them. Such modules should be available as a separate package.

What you can do:

* publish AMQP-Interop into a package
* add this package to Addons section
* (optionally) send PR to Codeception to AMQP.php mentioning an alternative module in documentation.

**Comment by makasim** [comment]
This PR was meant to be a replacement for current AMQP module. I don't touch the current one to make a transition simpler. The plan is 

* I am adding a new code
* then the current module is deprecated
* and removed. 

So, in the end, you have only one module.

**Comment by DavertMik** [comment]
Yeah, but I planned to remove AMQP module completely from the core - that's the plan.
As I said it was not so widely used it is not even tested well on our CI.
So I'd like to remove it as a dependency in next major version of Codeception.
There is no reason to keep everything in the core if we have Composer )

**Comment by makasim** [comment]
👍

**Pull request closed without merging**